File: core/src/main/java/com/zeyad/rxredux/core/eventbus/RxEventBusFactory.java
package com.zeyad.rxredux.core.eventbus;

import io.reactivex.BackpressureStrategy;

public final class RxEventBusFactory {

    private RxEventBusFactory() {
    }

    public static IRxEventBus getInstance() {
        return RxEventBus.getInstance();
    }

    public static IRxEventBus getInstance(BackpressureStrategy backpressureStrategy) {
        return RxEventBus.getInstance(backpressureStrategy);
    }
}

File: core/src/main/java/com/zeyad/rxredux/core/eventbus/RxEventBusFactory.kt
package com.zeyad.rxredux.core.eventbus

import io.reactivex.BackpressureStrategy

@Deprecated("Use your own implementation")
object RxIntentBusFactory {
    fun getInstance(backPressureStrategy: BackpressureStrategy = BackpressureStrategy.BUFFER):
            IRxIntentBus<Any> = RxIntentBus.getInstance(backPressureStrategy)
}
